src/netex-convertor/netexGenerator.ts

```diff
@@ -117,49 +117,47 @@ const netexGenerator = (
             coreData.nocCodeFormat;
-        resourceFrameToUpdate.responsibilitySets.ResponsibilitySet[0].roles.ResponsibilityRoleAssignment.ResponsibleOrganisationRef.$t = operatorPublicName;
+        resourceFrameToUpdate.responsibilitySets.ResponsibilitySet[0].roles.ResponsibilityRoleAssignment.ResponsibleOrganisationRef.$t =
+            coreData.operatorName;
         resourceFrameToUpdate.responsibilitySets.ResponsibilitySet[1].roles.ResponsibilityRoleAssignment.ResponsibleOrganisationRef.ref =
             coreData.nocCodeFormat;
-        resourceFrameToUpdate.responsibilitySets.ResponsibilitySet[1].roles.ResponsibilityRoleAssignment.ResponsibleOrganisationRef.$t = operatorPublicName;
+        resourceFrameToUpdate.responsibilitySets.ResponsibilitySet[1].roles.ResponsibilityRoleAssignment.ResponsibleOrganisationRef.$t =
+            coreData.operatorName;
+
         resourceFrameToUpdate.typesOfValue.ValueSet[0].values.Branding.id = coreData.brandingId;
-        resourceFrameToUpdate.typesOfValue.ValueSet[0].values.Branding.Name.$t = operatorPublicName;
+        // below 2 branding attributes, name and url, are new to point to point, but should not break anything
+        resourceFrameToUpdate.typesOfValue.ValueSet[0].values.Branding.Name.$t = coreData.operatorName;
         resourceFrameToUpdate.typesOfValue.ValueSet[0].values.Branding.Url.$t = coreData.website;
 
-        if (
-            ticket.type === 'multiOperator' &&
-            (isMultiOperatorGeoZoneTicket(ticket) || isSchemeOperatorTicket(ticket))
-        ) {
+        if (coreData.isMultiOperator && (isMultiOperatorGeoZoneTicket(ticket) || isSchemeOperatorTicket(ticket))) {
             const nocs = [...ticket.additionalNocs];
-            if (isMultiOperatorGeoZoneTicket(ticket)) {
-                nocs.push(ticket.nocCode);
+            if (coreData.isGeoZone) {
+                nocs.push((ticket as MultiOperatorGeoZoneTicket).nocCode);
                 resourceFrameToUpdate.organisations.Operator = getOrganisations(operatorData);
-            } else if (isBaseSchemeOperatorInfo(baseOperatorInfo) && isSchemeOperatorTicket(ticket)) {
+            } else if (isBaseSchemeOperatorInfo(baseOperatorInfo) && coreData.isSchemeOperator) {
                 resourceFrameToUpdate.organisations.Operator = getOrganisations(operatorData, baseOperatorInfo);
             }
             resourceFrameToUpdate.groupsOfOperators = getGroupOfOperators(operatorData);
-        } else if (ticket.type === 'multiOperator' && isMultiOperatorMultipleServicesTicket(ticket)) {
-            const nocs: string[] = ticket.additionalOperators.map(additionalOperator => additionalOperator.nocCode);
+        } else if (coreData.isMultiOperator && coreData.isMultiOperatorMultiServices) {
+            const nocs: string[] = ticket.additionalOperators.map(
+                (additionalOperator: { nocCode: any }) => additionalOperator.nocCode,
+            );
```

Comment:
Typescript complaining.

Reply:
Referencing the above comments (from line 142), I think this issue arises because the code doesn't know the exact structure of the data at this point. If we were to have a typeguard in place of the `isMultiOperator` or `isMultiOperatorMultiServices` attributes, this should fix the complaints.